pydantic/_internal/_decorators.py

```diff
@@ -208,6 +210,90 @@ def bind_to_cls(self, cls: Any) -> Decorator[DecoratorInfoType]:
         )
 
 
+def get_bases(tp: type[Any]) -> tuple[type[Any], ...]:
+    if is_typeddict(tp):
+        return tp.__orig_bases__  # type: ignore
```

Comment:
I see here https://github.com/python/cpython/pull/103698/files it should be a tuple as well. Could we do something like `typing.cast` here instead of `type: ignore`?

The type error here is likely caused by the fact that `type[Any]` doesn't necessarily  have `__orig_bases__` attr. Shall we add an assert here or cast `tp` then?

A `TypeGuard` on `is_typeddict` would help. Shall we add a todo here and contribute the `TypeGuard` to `typing_extensions`?

Reply:
Not that simple: https://github.com/microsoft/pyright/issues/3442.
A type ignore here is by far the easiest solution.